client/test_data/up_or_down_final_cards.json

```diff
@@ -4,71 +4,133 @@
         "location": "discard",
         "suitIndex": 0,
         "rank": 4,
-        "possibleCardsFromObservation": [
```

Comment:
if we keep changing the baseline, this test is a lot less useful. this test is to make sure changes like this one don't break existing logic, so we can't keep changing the baseline every time.

Reply:
I get that, although I'm not sure what I should be doing instead in this case...  I certainly wasn't setting out to lose the check we had before.  I don't think you're suggesting I need to make a change in this particular instance, if I understand correctly.  Just saying "we gotta stop doing this all the time"?

With the full deduction "check if it's impossible to be this card", there's not really a count of each suit/rank anymore.  It's either possible or it's not.  So I don't really have access to / a need for a count.  If we added some inference back in (perhaps for performance) and kept deduction then that'd probably be a third layer, in between `possibleCardsFromClues` and `possibleCardsFromDeduction`, with a map of suit/rank to count.

One option would be to still include the `possibleCardsFromObservation` assertion values from prior to this commit and in code convert it to a suit/rank list to assert against `possibleCardsFromDeduction`, so we can see which counts changed from non-zero to zero but that feels a little clunky to be carrying around counts if the logic isn't differentiating between the non-zero values.

Reply:
I think we can safely resolve this. It's unfortunate that the test isn't really helping us for this change but I think that's natural given that we changed the data structure and the test cares about the structure of the output.